### ISSUE TITLE
Stackoverflow exception when implementing IAuthenticationService

### ISSUE THREAD

**Issue by mattcowen** [OrchardCMS/Orchard#6192]
In the latest 1.9.x, when implementing the interface IAuthenticationService in a custom module Autofac generates a stackoverflowexception. The first occurrence seems to be in WorkContextImplementation on line 15.

```_workContextStateProviders = componentContext.Resolve<IEnumerable<IWorkContextStateProvider>>();```

I tried to add the following to my implementation of IAuthenticationService but it didn't make a difference.

Does anyone else see this?

``` [OrchardSuppressDependency("Orchard.Security.Providers.FormsAuthenticationService")] ```

**Comment by jtkech** [comment]
@mattcowen 

Try to use **Lazy** `IWorkContextStateProvider` as already done in the dev branch

Best

**Comment by mattcowen** [comment]
Thank you @jtkech. It has solved the exception. Is there a plan for applying this as a fix for 1.9.x or are we waiting on something else?

**Comment by jtkech** [comment]
@mattcowen good to know it can solve your issue.

The 1.9.x branch is periodically merged in the dev branch, but, normally, what is in the dev branch will be used for the next 1.10 release...

But, if @sebastienros agree, i can do a PR that target 1.9.x.

Best
